game/board.cpp

```diff
@@ -41,11 +39,13 @@ void Board::printBoard()
 
 // Regarde l'emplacement de la carte dans le Board et ajoute +1
 void Board::addCard(EstablishmentCard* card) {
-    cardsDecks.at(card)++;
+    if (cardsDecks.count(card))
+        cardsDecks.at(card)++;
+    else cardsDecks.insert({card, 1});
 }
 
 void Board::removeCard(EstablishmentCard* card) {
-    if (cardsDecks.at(card) != 0) {
+    if (auto search = cardsDecks.find(card); search != cardsDecks.end()) {
```

Comment:
ca fonctionne ça ?

Reply:
en fait faire .at(card) sur une carte qui n'existe pas dans la map ça throw
la méthode .find(card) renvoie un itérateur arrêté sur la card ou le .end() si pas trouvé

Reply:
tu veux pas remonter le find juste avant le if ?

Reply:
oui j'ai découvert ça dans des exemples sur stackoverflow, c'est pour limiter le scope de la variable qui ne sera pas réutilisée